packages/api-client/src/APIClient.ts

```diff
@@ -199,10 +210,56 @@ export class APIClient extends EventEmitter {
         service: new ServiceAPI(this.transport.http),
         team: new TeamAPI(this.transport.http),
       },
-      user: new UserAPI(this.transport.http),
+      user: new UserAPI(this.transport.http, backendFeatures),
+    };
+  }
+
+  /**
+   * Will compute all the capabilities of the backend API according to the selected version and the version response payload
+   * @param backendVersion The agreed used version between the client and the backend
+   * @param responsePayload? The response from the server
+   */
+  private computeBackendFeatures(backendVersion: number, responsePayload?: BackendVersionResponse): BackendFeatures {
+    return {
+      federationEndpoints: backendVersion > 0,
+      isFederated: responsePayload?.federated || false,
     };
   }
 
+  /**
+   * Will set the APIClient to use a specific version of the API (by default uses version 0)
+   * It will fetch the API Config and use the highest possible version
+   * @param acceptedVersions Which version the consumer supports
+   * @return The highest version that is both supported by client and backend
+   */
+  async useVersion(acceptedVersions: number[]): Promise<number> {
+    if (acceptedVersions.length === 1 && acceptedVersions[0] === 0) {
+      // Nothing to do since version 0 is the default one
+      return 0;
+    }
+    let backendVersions: BackendVersionResponse = {supported: [0]};
+    try {
+      backendVersions = await (
+        await this.transport.http.sendRequest<BackendVersionResponse>({url: '/api-version'})
+      ).data;
+    } catch (error) {}
+    const highestCommonVersion = backendVersions.supported
+      .sort()
+      .reverse()
+      .find(version => acceptedVersions.includes(version));
+
+    if (highestCommonVersion === undefined) {
+      throw new Error(
+        `Backend does not support requested versions [${acceptedVersions.join(
+          ',',
+        )}] (supported versions ${backendVersions.supported.join(',')})`,
```

Comment:
probably a silly change but just thinking this would be more readable with ', '

ex: 1,2,3,4
suggestion: 1, 2, 3, 4 

although it is for an error so who cares. if you like to add it feel free. 

```suggestion
          ', ',
        )}] (supported versions ${backendVersions.supported.join(', ')})`,
```